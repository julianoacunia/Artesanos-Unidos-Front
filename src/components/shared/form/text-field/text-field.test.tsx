import * as React from 'react';
import renderer from 'react-test-renderer';
import TextField from './text-field';

describe('Submit button', () => {
  const dispatch = jest.fn();
  test('should match snapshot and styles', () => {
    const tree = renderer
      .create(
        <TextField
          maxLength={10}
          input={{
            value: 'test',
            onBlur: () => ({}),
            onChange: () => ({}),
            name: 'test',
            onDragStart: () => ({}),
            onDrop: () => ({}),
            onFocus: () => ({})
          }}
          meta={{
            autofilled: false,
            asyncValidating: false,
            dirty: false,
            dispatch,
            form: 'test',
            initial: 'test',
            invalid: false,
            pristine: false,
            submitting: false,
            submitFailed: false,
            touched: false,
            valid: false,
            visited: false,
          }}
          testId="test"
          multiline={false}
          type="text"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});