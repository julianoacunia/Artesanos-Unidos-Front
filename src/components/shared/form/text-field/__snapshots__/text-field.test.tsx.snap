// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit button should match snapshot and styles 1`] = `
<div
  className="textFieldGroup"
>
  <div
    className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1"
    maxLength={10}
    onDragStart={[Function]}
    onDrop={[Function]}
    test-id="test"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        maxLength={10}
        name="test"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="test"
      />
    </div>
  </div>
  <div
    className="textFieldGroupError"
  />
</div>
`;
